pkg/manager/impl/validation/validation_test.go

```diff
@@ -334,3 +358,42 @@ func TestValidateOutputData(t *testing.T) {
 		assert.Equal(t, codes.ResourceExhausted, err.(errors.FlyteAdminError).Code())
 	})
 }
+
+func TestValidateDatetime(t *testing.T) {
+	t.Run("no datetime", func(t *testing.T) {
+		assert.NoError(t, ValidateDatetime(nil))
+	})
+	t.Run("datetime with valid format and value", func(t *testing.T) {
+		assert.NoError(t, ValidateDatetime(&core.Literal{
+			Value: &core.Literal_Scalar{
+				Scalar: &core.Scalar{
+					Value: &core.Scalar_Primitive{
+						Primitive: &core.Primitive{
+							Value: &core.Primitive_Datetime{Datetime: timestamppb.Now()},
+						},
+					},
+				},
+			},
+		}))
+	})
+	t.Run("datetime with valid format and value", func(t *testing.T) {
+		timestamp, err := time.Parse(time.RFC3339, "-1000000000-01-01T00:00Z") // its value is actually 0001-01-01 00:00:00 +0000
+		if err != nil {
+			//panic(err)
+		}
+
+		assert.NoError(t, ValidateDatetime(&core.Literal{
+			Value: &core.Literal_Scalar{
+				Scalar: &core.Scalar{
+					Value: &core.Scalar_Primitive{
+						Primitive: &core.Primitive{
+							Value: &core.Primitive_Datetime{Datetime: timestamppb.New(timestamp)},
+						},
+					},
+				},
+			},
+		}))
+	})
+	// cannot generate invalid timestamp with wrong format or e.g. too big month or year.
```

Comment:
one option here is to generate the invalid timestamp proto from your java application, dump this to json and save that json blob as a test file in this repo. Then, to implement this test case you can parse the proto from json using [jsonpb Unmarshal](https://pkg.go.dev/github.com/golang/protobuf/jsonpb) and validate that

Reply:
Thanks for the suggestion! We finally found out we could do this using timestamppb type directly, constructed using Seconds+Nanos.

Reply:
even better, awesome!